src/Interpreters/ProcessList.cpp

```diff
@@ -325,15 +319,21 @@ ProcessList::EntryPtr ProcessList::insert(
             std::move(thread_group),
             query_kind,
             settings,
-            watch_start_nanoseconds);
+            watch_start_nanoseconds,
+            is_internal);
 
         auto process_it = processes.emplace(
             processes.end(),
             query);
 
-        CancellationChecker::getInstance().appendTask(query, query_context->getSettingsRef()[Setting::max_execution_time].totalMilliseconds(), query_context->getSettingsRef()[Setting::timeout_overflow_mode]);
+        /// We should not include internal queries for limiting the number of simultaneous queries.
+        if (!is_internal)
+        {
+            ++non_internal_processes;
```

Comment:
It's unclear to me which mutex protects this variable (or query_kind_amounts). I don't see a mutex here, and they are not atomic